Add connection with Postgre

marl0vv opened the pull request:
- Add connection to Postgre
- Add ConversionTask model that represents table conversion_tasks in Post
- Add ConversionTaskRepository that manages interaction with database
- Add ConvertImageAsync that writes conversion tasks in database

marl0vv committed:
Add ConversionTask model

Add ConversionTask for a conversion_tasks table in a database

marl0vv committed:
Add Repository for conversion tasks

marl0vv committed:
Add method to add conversion request in database

Add convertImageAsync that adds conversion task to database

RavenVSS commented on src/main/java/com/glamik/webpconverter/controllers/ConverterController.java:
```diff
@@ -14,12 +17,15 @@
 
 @RestController
 @RequiredArgsConstructor
+//@AllArgsConstructor
```
https://projectlombok.org/features/constructor

Тут достаточно только `@RequiredArgsConstructor` использовать

RavenVSS commented on src/main/java/com/glamik/webpconverter/model/ConversionTask.java:
```diff
@@ -0,0 +1,30 @@
+import java.util.UUID;
+
+@Entity
+@Table(name = "conversion_tasks")
```
Название не обязательно указывает, если оно совпадает с названием класса. Spring сам в snake-case его переведёт

RavenVSS commented on src/main/java/com/glamik/webpconverter/model/ConversionTask.java:
```diff
@@ -0,0 +1,30 @@
+@Entity
+@Table(name = "conversion_tasks")
+@Getter @Setter
+@RequiredArgsConstructor
```
`@AllArgsConstructor`

marl0vv replied:
Сделал поля taskStatus и fileName final. В самой программе они меняться уже не будут, а будут перезаписаны в БД.

RavenVSS commented on src/main/java/com/glamik/webpconverter/model/ConversionTask.java:
```diff
@@ -0,0 +1,30 @@
+    private UUID id;
+
+    @Column(name = "task_status", nullable = false)
+    private String taskStatus;
```
Лучше сделать через enum

RavenVSS replied:
https://stackoverflow.com/questions/15988342/mapping-enum-to-string-in-hibernate

RavenVSS commented on src/main/java/com/glamik/webpconverter/controllers/ConverterController.java:
```diff
@@ -55,6 +61,26 @@ public ResponseEntity<PathResource> convertImage(@RequestParam("image") Multipar
+
+
+        ConversionTask conversionTask = conversionTaskRepository
+                .save(new ConversionTask("PROCESSING", originalFilename));
```
Вынести в сервис

RavenVSS commented on src/main/java/com/glamik/webpconverter/controllers/ConverterController.java:
```diff
@@ -55,6 +61,26 @@ public ResponseEntity<PathResource> convertImage(@RequestParam("image") Multipar
     }
 
+    @PostMapping("/convert-to-webp/async")
+    public ResponseEntity<UUID> convertImageAsync(@RequestParam("image") MultipartFile imageFile) {
```
nit: В отдельный контроллер

marl0vv committed:
Add enum ConversionTaskStatus

Add enum ConversionTaskStatus. Now taskStatus can be in only three states.

Also some formatting changes and renaming changes

marl0vv committed:
Create separate controller for async tasks

marl0vv committed:
Take out async method in a separate Controller

Create new controller for asyns method. Take out methods for working with file names to FileUtils class and make the public static.

marl0vv committed:
Take out saving conversion tasks to db into service

Work with repository is now in service

marl0vv committed:
Add UtilityClass annotation

marl0vv committed:
Change ConversionTask model to use RequiredArgsConstructor

marl0vv committed:
Fix minor stuff about formatting and naming

marl0vv committed:
Take out work with conversion tasks to a separate Service
